src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerSessionMonitor.cs

```diff
@@ -259,16 +259,23 @@ private void StartStream(int id, ProfilerSession session)
                 CancellationTokenSource targetToken;
                 if (monitoredCancellationTokenSources.TryGetValue(id, out targetToken))
                 {
-                    SendStoppedSessionInfoToListeners(session.XEventSession.Id);
-                    ProfilerSession tempSession;
-                    RemoveSession(session.XEventSession.Id, out tempSession);
+                    stopSessionError(session.XEventSession.Id);
                 }
             }, TaskContinuationOptions.OnlyOnFaulted);
 
             this.monitoredCancellationTokenSources.Add(id, threadCancellationToken);
             session.enableStreamLock();
         }
 
+        /// <summary>
+        /// Helper function for firing error and stopping session in case the session is stopped on the server. This is public for tests.  
+        /// </summary>
+        public void stopSessionError(int Id){
```

Comment:
C# convention is capitalized method names (`StopSessionError`)

A couple suggestions : 

1. Just call this StopSession or something similar
2. Remove the mention about it being "public for tests". That isn't really important for people to know. Just describe what it's actually accomplishing (which in reality is "sending notification to listeners and then removing the session from our lists")